src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -253,6 +261,51 @@ private void readRcaEnabledFromConf() {
             rcaEnabled = rcaEnabledDefaultValue;
           }
         });
+
+    // If RCA is enabled, update Analysis graph with Muted RCAs value
+    if (rcaEnabled) {
+      LOG.debug("Updating Analysis Graph with Muted RCAs");
+      readAndUpdateMutesRcas();
+    }
+  }
+
+  /**
+   * Reads the mutedRCAList value from the rca.conf file, performs validation on the param value
+   * provided and on successful validation, updates the AnalysisGraph with muted RCA value.
+   *
+   * <p>In case all the RCAs in param value are incorrect, return without any update.
+   */
+  private void readAndUpdateMutesRcas() {
+    // If the rca config file has been updated since the lastModifiedTimeInMillisInMemory in memory,
+    // refresh the `muted-rcas` value from rca config file.
+    long lastModifiedTimeInMillisOnDisk = rcaConf.getLastModifiedTime();
+    if (lastModifiedTimeInMillisOnDisk > lastModifiedTimeInMillisInMemory) {
+      try {
+        Set<String> rcasForMute = new HashSet<>(rcaConf.getMutedRcaList());
+        LOG.info("RCAs provided for muting : {}", rcasForMute);
+
+        Set<String> graphNodeNames = new HashSet<>();
+        RcaUtil.getAnalysisGraphComponents(rcaConf).forEach(
+                connectedComponent -> graphNodeNames.addAll(connectedComponent.getNodeNames()));
```

Comment:
I don't get it, why can't we just call `getNodeNames()` instead of calling the expensive method `getAnalysisGraphComponents` ?

Reply:
Agreed, so now we are maintaining a set which gets populated at the time of graph creation and is accessible via `getNodeNames()`